qutip/core/data/creator_util.py

```diff
@@ -0,0 +1,41 @@
+"""
+Define functions to use as Data creator for `create` in `convert.py`.
+"""
+
+from scipy.sparse import isspmatrix_csr, issparse
+import numpy as np
+from .csr import CSR
+from .base import Data
+from .dense import Dense
+
+__all__ = ['is_data', 'is_nparray', 'data_copy', 'np2dense', 'sparse2csr',
+           'isspmatrix_csr', 'issparse']
+
+
+def is_data(arg):
+    return isinstance(arg, Data)
+
+
+def is_nparray(arg):
+    return isinstance(arg, np.ndarray)
+
+
+def data_copy(arg, shape):
+    return arg.copy()
+
+
+def np2dense(arg, shape):
+    return Dense(arg, shape=shape, copy=False)
+
+
+def sparse2csr(arg, shape):
+    return CSR(arg.tocsr(), shape=shape)
+
+
+def true(arg):
+    return True
+
+
+def numpy_array(arg, shape):
+    array = np.array(arg, dtype=np.complex128)
+    return Dense(array, shape=shape, copy=False)
```

Comment:
I think the `np2dense`, `sparse2csr` and `numpy_array` functions are all unnecessary - the `CSR` and `Dense` constructors have all this behaviour already (and logically all of these _should_ copy the data).

Reply:
I though `CSR` only accepted csr input and `Dense`, array...
`Dense`'s copy default is `True` but `CSR`'s is `False`, I will add a copy option to the creator function.

Reply:
Oh, let's change `CSR.__init__` to `copy=True` - I think it's much safer if that's always the default.  Yeah, having a `copy` parameter on the `create` function is definitely the right call.  I think we'll need to change `Qobj.__init__` a little to cope, but that's ok - I think I knew we'd have to revisit that at some point anyway.